packages/react-vapor/src/components/validation/hoc/NonEmptyInputValidationHOC.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import * as _ from 'underscore';
+import {IDispatch} from '../../../utils/ReduxUtils';
+import {IInputOwnProps} from '../../input/Input';
+import {ValidationActions} from '../ValidationActions';
+import {ValidationIds} from '../ValidationIds';
+import {InferableComponentEnhancer} from './connectHOC';
+
+export interface INonEmptyInputValidationProps {
+    validationMessage?: string;
+    resetErrorOnUnmount?: boolean;
+}
+
+const mapDispatchToProps = (dispatch: IDispatch) => ({
+    setError: (id: string, error: string) => dispatch(ValidationActions.setError(id, error, ValidationIds.nonEmpty)),
+    clearError: (id: string) => dispatch(ValidationActions.clearError(id)),
+});
+
+export const nonEmptyInputValidationHOC = <T extends IInputOwnProps>(
```

Comment:
I would suggest changing the name for `withValueInput` or something in the same vein, the `with` makes it clear that we compose behavior